datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -688,6 +688,29 @@ fn build_predicate_expression(
                 return Ok(unhandled);
             }
         }
+        Expr::InList {
+            expr,
+            list,
+            negated,
+        } => {
+            if !list.is_empty() {
```

Comment:
I also recommend putting a limit on the size of the INLIST when the rewrite occurs otherwise we'll likely explode in evaluation  -- for example don't allow `IN (1,2,.......1000000000)`  -- maybe a limit if 10 or 20 would be good?

Reply:
Sounds reasonable!